### ISSUE TITLE
Allow custom exception handler

### ISSUE THREAD

**Issue by leastprivilege** [IdentityServer/IdentityServer3#2012]
* receives exception on unhandled exceptions we hand you the exception
* allows setting response status code, content (if any) and content type

For UI paths this needs to be called from the exception filter. For API paths we don't have an exception filter yet - investigate if filter or IExceptionHandler is more appropriate.

**Comment by omerlh** [comment]
Hey,
Any updates regarding this?
I need it also. At least in my case (which does not involve ui path) IExceptionHandler would be enough, but I am not sure this is the case in this feature. If so, I can create a PR.

**Comment by leastprivilege** [comment]
What is your use case?

**Comment by omerlh** [comment]
Hey
I just want to customize the 500 errors - today is send back the stacktrace (unless it is something I've cahnged) and I wanted to just log and send generic message.

**Comment by leastprivilege** [comment]
A stacktrace? I think this is only for connections from localhost. Did you try it with a remote server already?

**Comment by omerlh** [comment]
Nope, but if this is the case than this is more than enough. I will update if I noticed otherwise.

**Comment by mclausen** [comment]
Hi Idsv team! Whats is the status on this?

We also this feature badly. In our scenario we want to be able to return  a custom response given a certain type of exception is been thrown from inside idsv. In normal WebApi you would just create and IExceptionFilter and hook it up into HttpConfiguration. It would be neat if idsv provided same functionality.

**Comment by leastprivilege** [comment]
The status is that we don't have enough dev resources to make this feature happen right now.
